src/Fusion.Resources.Functions.Common/ApiClients/SummaryApiClient.cs

```diff
@@ -3,33 +3,81 @@
 
 namespace Fusion.Resources.Functions.Common.ApiClients;
 
-
 public class SummaryApiClient : ISummaryApiClient
 {
     private readonly HttpClient summaryClient;
 
+    private readonly JsonSerializerOptions jsonSerializerOptions = new JsonSerializerOptions
+    {
+        PropertyNameCaseInsensitive = true
+    };
+
     public SummaryApiClient(IHttpClientFactory httpClientFactory)
     {
         summaryClient = httpClientFactory.CreateClient(HttpClientNames.Application.Summary);
         summaryClient.Timeout = TimeSpan.FromMinutes(2);
     }
 
-    public async Task PutDepartmentsAsync(IEnumerable<ApiResourceOwnerDepartments> departments,
+    public async Task PutDepartmentAsync(ApiResourceOwnerDepartment department,
         CancellationToken cancellationToken = default)
     {
-        var parallelOptions = new ParallelOptions()
-        {
-            CancellationToken = cancellationToken,
-            MaxDegreeOfParallelism = 10,
-        };
+        using var body = new JsonContent(JsonSerializer.Serialize(department, jsonSerializerOptions));
+
+        // Error logging is handled by http middleware => FunctionHttpMessageHandler
+        using var _ = await summaryClient.PutAsync($"departments/{department.DepartmentSapId}", body,
+            cancellationToken);
+    }
+
+    public async Task<ICollection<ApiResourceOwnerDepartment>?> GetDepartmentsAsync(
+        CancellationToken cancellationToken = default)
+    {
+        using var response = await summaryClient.GetAsync("departments", cancellationToken);
+
+        if (!response.IsSuccessStatusCode)
+            return null;
+
+        await using var contentStream = await response.Content.ReadAsStreamAsync(cancellationToken);
 
-        await Parallel.ForEachAsync(departments, parallelOptions, async (ownerDepartments, token) =>
+        return await JsonSerializer.DeserializeAsync<ICollection<ApiResourceOwnerDepartment>>(contentStream,
+                   jsonSerializerOptions,
+                   cancellationToken: cancellationToken)
+               ?? Array.Empty<ApiResourceOwnerDepartment>();
+    }
+
+    public async Task<ApiWeeklySummaryReport?> GetLatestWeeklyReportAsync(string departmentSapId,
+        CancellationToken cancellationToken = default)
+    {
+        var lastMonday = GetCurrentOrLastMondayDate();
```

Comment:
Could it be a point do add some comments? To bring some context to the code?

Reply:
Added and updated some code comments